---
title: "DASE user's guide"
author: |
  | Xiang Liu <xiang.liu@moffitt.org>
  | Mingxiang Teng <mingxiang.teng@moffitt.org>
  | Department of Biostatistics and Bioinformatics, H. Lee Moffit Cancer Center &
  | Research Institute, Tampa, FL, USA
date: "`r Sys.Date()`"  
output: 
  rmarkdown::html_vignette:
    toc: true
graphics: yes
vignette: >
  %\VignetteIndexEntry{DASE user's guide}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignettePackage{DASE}
  %\VignetteEncoding{UTF-8}
---

```{r, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  error = FALSE,
  warning = FALSE, 
  message = FALSE,
  cache = FALSE,
  comment = "#>",
  dev="png",
  fig.show = "hold",
  fig.width=4,
  fig.height=4,
  fig.align="center"
)
```

# Introduction

Super enhancers (SEs) were proposed as broad regulatory domains on genome, usually spanning a minimum
of thousands of base pairs and consisting of multiple constitute enhancers. The constitute enhancers work together as a unit, instead of separately, to facilitate high enhancer activity. Aberrant SE activities, 
which are critical to understand disease mechanisms, could be raised by the alterations of one or more of 
their constitute enhancers. However, the state-of-art binary strategy in calling differential SEs only relies on overall activity changes, neglecting the local dynamics of constitute enhancers within SEs. DASE uses a weighted spline model to identify differential SEs from two conditions by accounting for the combinatorial effects of constitute enhancers weighted with their activities and locations (internal dynamics). In addition to overall changes, our medthod finds four novel types (*Shortened/lengthened*, *shifted*, *hollowed* and *other complex scenarios*) of differential SEs pointing to the structural differences within SEs.

# Getting Started

Load the package in R

```{r library}
library(DASE)
```

# Preparing Inputs

DASE requires the following input files:

  1. enhancer bed file: a merged enhancer bed file includes the enhancer peaks of all samples.
  1. SE bed file: a merged SE bed file includes the SE regions of all samples.
  1. Coverage files: can either be the path of bam/bw files for each condition and replicate, or just an enhancer count table of all conditions and replicates.

Following are the examples of enhancer and SE bed files

## *Enhancer bed file*

The enhancer file can be any format of bed files, just make sure the first 6 columns contains the information of "chr","start","end","name","score",and "strand". Here is an example of bed files with only 6 columns.

```{r enhancer_data}
# load enhancer bed file
enhancer_path <- system.file("extdata","enhancer.bed",package="DASE")
enhancer_region <- read.table(enhancer_path,sep="\t",header=F)
head(enhancer_region)
```

## *SE bed file*

The SE file can be any format of bed files, just make sure the first 6 columns contains the information of "chr","start","end","name","score",and "strand". Here is an example of bed files with only 6 columns.

```{r SE_data}
# load SE bed file
se_path <- system.file("extdata","SE.bed",package="DASE")
se_region <- read.table(se_path,sep="\t",header=F)
head(se_region)
```

# Basic usage of DASE

In this section, we use DASE to find differential SEs with internal dynamics by comparing human chromosome 21 of two cancer cell lines (K562 and MCF7). Here, we focus on the basic usage of running DASE with different coverage input files (BAM, BigWig, and enhancer raw count table). This default setting will not include enhancer blacklist which indicates regions that don't have enhancers. In addition, under default setting, DASE will run permutation 10 times to get a significant threshold to identify SE categories. More additional features can be found later in the "Additional options" section.

## *Run DASE with BAM or BigWig coverage files*

This section shows you how to use DASE with BAM or BigWig coverage files. The BAM or BigWig files are for the reads count of enhancers in different samples. DASE uses *featureCounts* to get the enhancer abundance of each sample with BAM file and *rtracklayer* with BigWig file. The first step is to get the path of BAM or BigWig file for each samples. Then with the imported enhancer and SE region files, we can run DASE as follow.

```{r DASE1}
# path of BAM file for each condition
s1_r1_bam <- system.file("extdata","K562_1_chr21.bam",package="DASE")
s1_r2_bam <- system.file("extdata","K562_2_chr21.bam",package="DASE")
s2_r1_bam <- system.file("extdata","MCF7_1_chr21.bam",package="DASE")
s2_r2_bam <- system.file("extdata","MCF7_2_chr21.bam",package="DASE")

# running DASE with BAM files
se_region$V2
test_1 <- SEfilter(se_in = se_region)
DASE_out <- DASE(se_region,enhancer_region,
                 s1_r1_bam=s1_r1_bam,s1_r2_bam=s1_r2_bam,
                 s2_r1_bam=s2_r1_bam,s2_r2_bam=s2_r2_bam)
```

Run DASE with BigWig files.

```{r DASE2}
# path of BigWig file for each condition
s1_r1_bw <- system.file("extdata","K562_1_chr21.bw",package="DASE")
s1_r2_bw <- system.file("extdata","K562_2_chr21.bw",package="DASE")
s2_r1_bw <- system.file("extdata","MCF7_1_chr21.bw",package="DASE")
s2_r2_bw <- system.file("extdata","MCF7_2_chr21.bw",package="DASE")

# running DASE with BigWig files
DASE_out <- DASE(se_in=se_region,e_in=enhancer_region,data_type = "bw",
                 s1_r1_bam=s1_r1_bw,s1_r2_bam=s1_r2_bw,
                 s2_r1_bam=s2_r1_bw,s2_r2_bam=s2_r2_bw)

```

## *Run DASE with enhancer raw count table*

This section shows you how to use DASE with enhancer raw count table. The format of count table is shown bellow. Here we don't need BAM or BigWig files, because we already have the enhancer counts for each sample. We can run DASE as follow. This step will skip *featureCount*.

```{r DASE3}
# read enhancer count table
enhancer_count_path <- system.file("extdata","chr21_enhancer_count.txt",package="DASE")
enhancer_count <- read.table(enhancer_count_path,sep="\t",header=T)
head(enhancer_count)

# run DASE
DASE_out_count <- DASE(se_in=se_region,e_in=enhancer_region,
                 enhancer_count_table=enhancer_count)

```

# Interpretation of DASE output files

The output of DASE is a list with multiple data types including:

  1. lfc_shrink: a shrinking lfc object from DESeq2, it can be used to get a MA plot
  1. cutoff: significant threshold of fitted log2 fold change.
  1. density_plot: a density plot of permutation and original fitted log2 fold change, if *permut=T*.
  1. boxplot: a boxplot of final SE categories
  1. se_category: a data frame contains final SE categories
  1. pattern_list: a list contains figures of each SE's pattern
  1. se_fit: a data frame contains DESeq2 output and spline-fitted log2 fold change of all constitute enhancers

  
## Significant threshold
We use permutation of spline fitted log2 fold change to decide the significant threshold. Under default settings, DASE will run permutation 10 times with *SEpermut* function. If don't use permutation, the default significant threshold is -1 and 1. You can use your own threshold with *cutoff_v* parameter. Please refer to function manual or *Additional options* section.

```{r threshold}
# Significant threshold
DASE_out_count$cutoff

# Permutation density plot
DASE_out_count$density_plot
```

Black lines indicates the threshold which is obtained by the inflection point.

## Super-enhancer internal dynamic categories

*SE_category, pattern_list, se_fit* are the output related to SE internal dynamics.

```{r se_cat_out}
# se_categories
head(DASE_out_count$se_category)
```
  
```{r se_pattern_example}
# an example of one sample
DASE_out_count$pattern_list[[1]]
```

This figure shows the constitute enhancer patterns within SE "chr21_34863697_34890719" which identified as *Shortened*. Black line is the fitted log2 fold change curve, dots indicate constitute enhancers. Red dots indicate the constitute enhancers with large weights.
  
```{r se_fit}
# example of se_fit
head(DASE_out_count$se_fit)
```
  
# Additional options
In addition of default parameters, DASE can take an blacklist file which contains the regions cannot be identified as enhancers. DASE also can take a customized blacklist region with *custom_range* parameter. We have included a blacklist file from ENCODE (accession ID: ENCFF356LFX) in our package.

```{r blacklist}
# blacklist file
blacklist_path <- system.file("extdata","region_blacklist.bed",package="DASE")
blacklist_region <- read.table(blacklist_path,sep="\t",header=F)
head(blacklist_region)
```

There is also an option whether to choice permutation or not. The default setting of DASE will run permutation 10 times. You can turn it off with *permut=F*. When there is no permutation, the default threshold is -1 and 1. You can chose any threshold you like with *cutoff_v* parameter. However, defining customized threshold is only available under *permut=F*. 

## *DASE with enhancer blacklist region*
Example of using blacklist options. Here, we focus on using DASE with enhancer count table.
```{r blacklist_run}
# run DASE with blacklist file and customized region
DASE_out_bl <- DASE(se_in=se_region,e_in=enhancer_region,bl_file = blacklist_region,
                 custom_range = c("chr21:14735763-29634779","chr21:33539902-43710703"),
                 enhancer_count_table=enhancer_count)

```

You can find that the number of SEs (8) in "DASE_out_bl" is less than "DASE_out_count" which is without blacklist region (34).

## *DASE with no permutation*
Some examples of using permutation options. Here, we focus on using DASE with enhancer count table.

```{r permut_1}
# run DASE with permutation 3 times
DASE_out <- DASE(se_in=se_region,e_in=enhancer_region,times=3,
                 cutoff_v = c(-2,2),enhancer_count_table=enhancer_count)
```

```{r no_permut}
# run DASE with no permutation and customized threshold
DASE_out_p <- DASE(se_in=se_region,e_in=enhancer_region,permut = F,
                 cutoff_v = c(-3,3),enhancer_count_table=enhancer_count)

# boxplots of different threshold
DASE_out_count$boxplot
DASE_out_p$boxplot
```

Because we have a larger threshold (bottom plot), more SEs are identified as *similar* category than before (top plot).

# Citation
If you used DASE, please cite our paper: [https://www.biorxiv.org/content/10.1101/2021.09.25.461810v1] 

